src/language/pythonLanguage.ts

```diff
@@ -0,0 +1,77 @@
+import { buildParameterTypeLinksFromMatches } from './helpers.js'
+import { Language } from './types.js'
+
+export const pythonLanguage: Language = {
+  defineParameterTypeQueries: [
+    `
+(call
+  function: (identifier) @method
+  arguments: (argument_list
+  (keyword_argument
+       name: (identifier) @name-key
+       value: (string) @name
+  )
+  (keyword_argument
+       name: (identifier) @regexp-key
+       value: (string) @expression
+  )
+  )
+  (#eq? @method "ParameterType")
+  (#eq? @name-key "name")
+  (#eq? @regexp-key "regexp")
+) @root
+`,
+  ],
+  defineStepDefinitionQueries: [
+    `
+(decorated_definition
+    (decorator
+        (call
+            function: (identifier) @method
+            arguments: (argument_list (string) @expression)
+        )
+    )
+    (#match? @method "(given|when|then)")
+) @root
+`,
+  ],
+
+  convertParameterTypeExpression(expression: string) {
+    return toStringOrRegExp(expression)
```

Comment:
You should throw an exception here. Like the PHP version does: https://github.com/cucumber/language-service/blob/8d3e3cb268e34421fea94105391ad464da3157ec/src/language/phpLanguage.ts#L16-L18